### PR TITLE
set up store and install  plugin on mirage

### PR DESCRIPTION
**Pull request by Pierregvx**

### CODE REVIEW

**Review by pivilartisant — APPROVED** [review]

File: web/massastation/src/mirage/server.js [hunk 1]
```diff
@@ -79,7 +111,24 @@ function mockServer(environment = ENV.DEV) {
...
+          status: 'Up',
+          updatable: false,
+          name: storePlugin?.name || faker.lorem.word(),
+          author: storePlugin?.author || faker.person.firstName(),
+          description: storePlugin?.description || faker.lorem.sentence(),
+          version: storePlugin?.version || faker.system.semver(),
+          home: storePlugin?.url || `/plugin/massalabs/${faker.lorem.word()}/`,
+        };
+        schema.plugins.create(newPlugin);
+        return newPlugin;
```
Comment (Thykof) [inline web/massastation/src/mirage/server.js:130]: I am not sure here https://github.com/massalabs/station/blob/main/api/swagger/server/restapi/resource/swagger.yml#L386 it returns a 204 no content
```suggestion
        return null;
```

**Review by Thykof — CHANGES_REQUESTED** [review]

File: web/massastation/src/mirage/server.js [hunk 2]
```diff
@@ -79,7 +111,24 @@ function mockServer(environment = ENV.DEV) {
...
+          status: 'Up',
+          updatable: false,
+          name: storePlugin?.name || faker.lorem.word(),
+          author: storePlugin?.author || faker.person.firstName(),
+          description: storePlugin?.description || faker.lorem.sentence(),
+          version: storePlugin?.version || faker.system.semver(),
+          home: storePlugin?.url || `/plugin/massalabs/${faker.lorem.word()}/`,
+        };
+        schema.plugins.create(newPlugin);
+        return new Response(200, {});
```
Comment (Thykof) [inline web/massastation/src/mirage/server.js:130]:
```suggestion
        return new Response(204, {});
```

**Review by Thykof — APPROVED** [review]

File: web/massastation/src/mirage/server.js [hunk 3]
```diff
@@ -79,7 +111,26 @@ function mockServer(environment = ENV.DEV) {
...
         return new Response(200, {});
       });
-
+      this.post('plugin-manager', (schema, request) => {
+        const sourceURL = request.queryParams.source;
+        const storePlugin = schema.stores.findBy(
+          (store) => store.file.url === sourceURL,
+        );
+        // use the parameters of the store plugin if it exists
+        // otherwise generate random values
```
Comment (mazmassa) [inline web/massastation/src/mirage/server.js:120]:
```suggestion
```

File: web/massastation/src/mirage/server.js [hunk 4]
```diff
@@ -79,7 +111,26 @@ function mockServer(environment = ENV.DEV) {
         });
         return new Response(200, {});
       });
-
+      this.post('plugin-manager', (schema, request) => {
+        const sourceURL = request.queryParams.source;
+        const storePlugin = schema.stores.findBy(
```
Comment (mazmassa) [inline web/massastation/src/mirage/server.js:116]:
```suggestion

        const storePlugin = schema.stores.findBy(
```

File: web/massastation/src/mirage/server.js [hunk 5]
```diff
@@ -79,7 +111,26 @@ function mockServer(environment = ENV.DEV) {
...
+          status: 'Up',
+          updatable: false,
+          name: storePlugin?.name || faker.lorem.word(),
+          author: storePlugin?.author || faker.person.firstName(),
+          description: storePlugin?.description || faker.lorem.sentence(),
+          version: storePlugin?.version || faker.system.semver(),
+          home: storePlugin?.url || `/plugin/massalabs/${faker.lorem.word()}/`,
+        };
+        schema.plugins.create(newPlugin);
+        return new Response(204);
```
Comment (mazmassa) [inline web/massastation/src/mirage/server.js:132]:
```suggestion
        };
        
        return schema.plugins.create(newPlugin);
```

File: web/massastation/src/mirage/server.js [hunk 6]
```diff
@@ -88,6 +139,10 @@ function mockServer(environment = ENV.DEV) {
 
         return plugin.destroy();
       });
+      this.get('plugin-store', (schema) => {
+        let { models: stores } = schema.stores.all();
+        return stores;
+      });
```
Comment (mazmassa) [inline web/massastation/src/mirage/server.js:145]:
```suggestion

      this.get('plugin-store', (schema) => {
        let { models: stores } = schema.stores.all();
        
        return stores;
      });
```

File: web/massastation/src/mirage/server.js (hunk 5 above)
Comment (Thykof) [reply web/massastation/src/mirage/server.js:132]: nope, see https://github.com/massalabs/station/pull/799#discussion_r1234159547

File: web/massastation/src/mirage/server.js (hunk 6 above)
Comment (Thykof) [reply web/massastation/src/mirage/server.js:145]: we should configure prettier to fix these blank lines

Comment (mazmassa) [reply web/massastation/src/mirage/server.js:145]: Totally agree! No doubts! BTW the console.log we add is not working neither!

File: web/massastation/src/mirage/server.js (hunk 5 above)
Comment (Pierregvx) [reply web/massastation/src/mirage/server.js:132]: wouldn't this return the plugin ?

Comment (mazmassa) [reply web/massastation/src/mirage/server.js:132]: According BE docs not. It's an empty 204 as you did! Good for me!

**Review by mazmassa — APPROVED** [review]
Left some small comments. LGTM!

**Review by mazmassa — APPROVED** [review]

**Pull request merged**